pontoon/sync/formats/base_json_file.py

```diff
@@ -0,0 +1,56 @@
+import codecs
+import json
+import logging
+
+from collections import OrderedDict
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+from pontoon.sync.exceptions import ParseError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.utils import create_parent_directory
+
+log = logging.getLogger(__name__)
+
+
+class JSONResource(ParsedResource):
+    @property
+    def translations(self):
+        return sorted(self.entities.values(), key=lambda e: e.order)
+
+    def open_json_file(self, path, SCHEMA, source_resource=None):
+        try:
+            with codecs.open(path, "r", "utf-8") as resource:
+                json_file = json.load(resource, object_pairs_hook=OrderedDict)
+                validate(json_file, SCHEMA)
+                return json_file
+
+        except (OSError, ValueError, ValidationError) as err:
+            # If the file doesn't exist or cannot be decoded,
+            # but we have a source resource,
+            # we can keep going, we'll just not have any translations.
+            if source_resource:
+                return
```

Comment:
if nothing is returned, the syncing process will crash along the way:

```
2022-02-09T13:22:27.591077+00:00 app[worker.1]: File "/app/pontoon/sync/formats/json_keyvalue.py", line 107, in traverse_json
2022-02-09T13:22:27.591078+00:00 app[worker.1]: for key, value in data.copy().items():
2022-02-09T13:22:27.591080+00:00 app[worker.1]: AttributeError: 'NoneType' object has no attribute 'copy'
```

Isn't returning an empty dict sufficient ? 
```python
except (OSError, ValueError, ValidationError) as err:
     # If the file doesn't exist or cannot be decoded,
     # but we have a source resource,
     # we can keep going, we'll just not have any translations.
         if source_resource:
             return {}

Reply:
Thank you :)
It seem that this will prevent the issue from happening on the web extension json parser too!